### DISCUSSION TITLE
Add option specifically to resolve modified-deleted merge conflicts

### DISCUSSION THREAD

**Discussion by Protonull** [jesseduffield/lazygit#3868] [Ideas]
If you are rebasing or merging BranchB onto BranchA, but BranchA has modified a file and BranchB has deleted that file (or vice versa), there doesn't seem to be a clear and idiot-proof way to resolve this. You can stash, discard, or reset the file, but it's not entirely clear what these would do. And attempting to `ENTER` the file, as you would with any other conflicted change, gets you a `Can only stage individual lines for tracked files` error. Perhaps `ENTER` could, in these kinds of scenarios, instead present a dialogue that lets you choose between the versions of the file similarly to Gitkraken: 
![image](https://github.com/user-attachments/assets/f8a1c467-b954-4af7-a939-1a121d957022)

**Comment by Saafo** [comment]
Same problem. Should we convert this to issue?

The temporary way I've found is `open external merge tool`(e.g. git mergetool) and select to delete or modify the file.

Btw, when I select delete in git mergetool and returns back to lazygit, it remains blank.

**Comment by stefanhaller** [comment]
I have run into this in the past too. I like the idea of presenting a dialog that lets you pick options what to do with the file. (However, I wonder what the "Keep Base Version" button is good for; I'd have thought that you only need two buttons, one for keeping the modified version of the file and one for deleting it.)

This is not the only thing that needs to be improved though: for a file that was modified locally and deleted in incoming, you want to see the diff of what you changed locally; because it is very likely that the incoming change moved the code somewhere else, so you'll manually have to apply your modifications to that other place. And vice versa, of course.

Planning to work on this soon, I happen to have some time this week. :-)

**Comment by stefanhaller** [comment]
Here's a PR if you want to test it: #4431
